stable_baselines3/common/noise.py

```diff
@@ -84,3 +85,55 @@ def reset(self) -> None:
 
     def __repr__(self) -> str:
         return f'OrnsteinUhlenbeckActionNoise(mu={self._mu}, sigma={self._sigma})'
+
+
+class VectorizedActionNoise(ActionNoise):
+    """
+    A Vectorized action noise for parallel environments.
+
+    :param base_noise: ActionNoise The noise generator to use
+    :param n_envs: (int) The number of parallel environments
+    """
+
+    def __init__(self, base_noise: ActionNoise, n_envs: int):
+        super().__init__()
+        self.n_envs: int = n_envs
+        self.noises = [copy.deepcopy(base_noise) for _ in range(n_envs)]
+
+    def reset(self, indices: Optional[Iterable[int]] = None) -> None:
+        """
+        Reset all the noise processes, or those listed in indices
+
+        :param indices: Optional[Iterable[int]] The indices to reset. Default: None.
+            If the parameter is None, then all processes are reset to their initial position.
+        """
+        if indices is None:
+            indices = range(len(self.noises))
+
+        for index in indices:
+            self.noises[index].reset()
+
+    def __repr__(self) -> str:
+        base_repr = repr(self.noises[0]) if len(self.noises) else ""
+        return f"VecNoise(BaseNoise={base_repr}), n_envs={len(self.noises)})"
+
+    def __call__(self) -> np.ndarray:
+        """
+        Generate and stack the action noise from each noise object
+        if n_envs == 0, returns a 0x0 matrix.
```

Comment:
why would you allow such thing?

Reply:
Because I wanted to discuss this. Ranges alone don't consider the length since `range(negative)` is always valid even if it never gives elements so I wanted to add an assertion but, in general, the parameters are assumed to be valid and sound. So I want to suggest to set a 'standard way to handle these cases', e.g. throwing `AssertionError` or `ValueError`.

Reply:
yes, please add an `assert n_envs > 0` in the constructor